spec/controllers/user/omniauth_callbacks_controller_spec.rb

```diff
@@ -5,24 +5,34 @@
 
   with_tenant(:instance) do
     before { controller.request.env['devise.mapping'] = Devise.mappings[:user] }
+    self::UID = SecureRandom.hex
 
     describe '#facebook' do
-      let(:facebook_data) { OmniAuth.config.mock_auth[:facebook].dup }
+      let(:uid) { SecureRandom.hex }
+      let(:facebook_data) do
+        data = OmniAuth.config.mock_auth[:facebook].dup
+        data.uid = uid
+        data
+      end
       before { controller.request.env['omniauth.auth'] = facebook_data }
       subject { get :facebook }
 
-      it 'creates a user identity' do
-        subject
-        identity = User::Identity.find_by(provider: facebook_data.provider, uid: facebook_data.uid)
-        expect(identity).to be_present
-      end
+      context 'when required data is provided' do
+        let(:uid) { self.class::UID }
```

Comment:
Oh, the data configured is only available in the feature tests. you should do this in controller test.